File: Sen/Count_Primes.py
class Solution(object):
    def countPrimes(self, n): # https://primes.utm.edu/howmany.html http://en.wikipedia.org/wiki/Sieve_of_Eratosthenes
        """
        :type n: int
        :rtype: int
        """
        isPrime = [True] * max(n, 2)
        isPrime[0], isPrime[1] = False, False
        x = 2
        while x * x < n:
        	if isPrime[x]:
        		p = x * x
        		while p < n:
        			isPrime[p] = False
        			p += x
        	x += 1
        return sum(isPrime)

n = 14
i = Solution()
print i.countPrimes(n)

File: xiaoming/1006/CountPrimes.java

public class CountPrimes {
	
	public int countPrimesLessN(int n) {
		if (n <= 2)
			return 0;
	 	boolean[] primes = new boolean[n];
		for (int i = 2; i < n; i++)
			primes[i] = true;
	 
		for (int i = 2; i < n; i++) {
			if (primes[i]) {
				for (int j = i * 2; j < n; j += i)
					primes[j] = false;
			}
		}
	 
		int count = 0;
		for (int i = 2; i < n; i++) {
			if (primes[i])
				count++;
		}
	 
		return count;
	}

	
//	public int countPrimes(int n) {	 
//	    ArrayList<Integer> primes = new ArrayList<Integer>();
//	 
//	    if(n<=1) 
//	        return 0;
//	    if(n==2)
//	        return 1;
//	    if(n==3)
//	        return 2;
//	 
//	    primes.add(2);
//	    primes.add(3);
//	 
//	    for(int i=4; i<n; i++){
//	        boolean isPrime = true;
//	        for(int p: primes){
//	            int m = i%p;
//	            if(m==0){
//	                isPrime = false;
//	                break;
//	            }
//	        }	 
//	        if(isPrime){
//	            primes.add(i);
//	        }
//	    }	 
//	    return primes.size();
//	}
}
